### PR TITLE
Fix required icon missing when hideRoot is true

### PR DESCRIPTION
**Pull request by Hypercubed**
## Summary

- Feature: support dynamic `hideRoot` inputs
- Fix: required icon not showing for top level when `hideRoot` is true

## Checklist

- [ ] \*Added unit tests
- [x] \*Added a code reviewer
- [x] Added changes to `/projects/swimlane/ngx-ui/CHANGELOG.md` under HEAD (Unreleased)
- [x] Updated the demo page
- [ ] Included screenshots of visual changes

_\*required_

### CODE REVIEW

File: projects/swimlane/ngx-ui/src/lib/components/json-editor/json-editor-flat/json-editor-node-flat/json-editor-node-flat.component.ts [hunk 1]
```diff
@@ -53,19 +53,19 @@ export class JsonEditorNodeFlatComponent extends JsonEditorNode implements OnIni
 
   requiredIndicator: SafeHtml;
 
-  constructor(public dialogMngr: DialogService, private domSanitizer: DomSanitizer) {
-    super(dialogMngr);
-    this.requiredIndicator = this.domSanitizer.bypassSecurityTrustHtml(requiredIndicatorIcon);
-  }
-
-  ngOnInit() {
+  get nextLevel() {
```
Comment (duredhel13) [inline projects/swimlane/ngx-ui/src/lib/components/json-editor/json-editor-flat/json-editor-node-flat/json-editor-node-flat.component.ts:56]: We only need to do this if/else once. No big deal in this case, but a getter is gonna keep doing this all the time.

Comment (duredhel13) [reply projects/swimlane/ngx-ui/src/lib/components/json-editor/json-editor-flat/json-editor-node-flat/json-editor-node-flat.component.ts:56]: No sure about naming this `nextLevel`. It's really the actual level. If something could be changed is the @Input from `level` to `previousLevel`.

Comment (Hypercubed) [reply projects/swimlane/ngx-ui/src/lib/components/json-editor/json-editor-flat/json-editor-node-flat/json-editor-node-flat.component.ts:56]: I can remove the getter and mode the logic to on-changes.  Eisting code did `this.level += 1` which alterend the input and made the result not reactive to changes in input.

Comment (duredhel13) [reply projects/swimlane/ngx-ui/src/lib/components/json-editor/json-editor-flat/json-editor-node-flat/json-editor-node-flat.component.ts:56]: Ok, it makes it we want to hide the root dynamically. I'd prefer `onChanges` but it's up to you.

**Pull request merged**